lib/dangermattic/plugins/ios_release_check.rb

```diff
@@ -0,0 +1,91 @@
+    #
+    # @return [void]
+    def check_core_data_model_changed
+      warning = 'Do not edit an existing Core Data model in a release branch unless it hasn\'t been released to testers yet. ' \
+                'Instead create a new model version and merge back to develop soon.'
+
+      common_release_checks.check_file_changed(
+        file_comparison: ->(path) { File.extname(path) == '.xcdatamodeld' },
+        message: warning,
+        on_release_branch: true
+      )
+    end
+
+    # Checks if any Localizable.strings file has been modified on a release branch, otherwise reporting a warning.
+    #
+    # @return [void]
+    def check_modified_localizable_strings_on_release
+      common_release_checks.check_file_changed(
+        file_comparison: ->(path) { File.basename(path) == LOCALIZABLE_STRINGS_FILE },
+        message: "The `#{LOCALIZABLE_STRINGS_FILE}` files should only be updated on release branches, when the translations are downloaded by our automation.",
+        on_release_branch: false
+      )
+    end
+
+    # Checks if the en.lproj/Localizable.strings file has been modified on a regular branch, otherwise reporting a warning.
+    #
+    # @return [void]
+    def check_modified_en_strings_on_regular_branch
+      common_release_checks.check_file_changed(
+        file_comparison: ->(path) { base_strings_file?(path: path) },
+        message: "The `#{BASE_STRINGS_FILE}` file should only be updated before creating a release branch.",
+        on_release_branch: true
+      )
+    end
+
+    # Checks if a translation file (*.lproj/Localizable.strings) has been modified on a release branch, otherwise reporting a warning.
+    #
+    # @return [void]
+    def check_modified_translations_on_release_branch
+      common_release_checks.check_file_changed(
+        file_comparison: ->(path) { !base_strings_file?(path: path) && File.basename(path) == LOCALIZABLE_STRINGS_FILE },
+        message: "Translation files `*.lproj/#{LOCALIZABLE_STRINGS_FILE}` should only be updated on a release branch.",
+        on_release_branch: false
+      )
+    end
+
+    # Checks if changes made to the release notes are also followed by changes in the App Store strings file.
+    #
+    # @return [void]
+    def check_release_notes_and_app_store_strings
+      common_release_checks.check_release_notes_and_store_strings(
+        release_notes_file: 'Resources/release_notes.txt',
+        po_file: 'Resources/AppStoreStrings.po'
+      )
+    end
+
+    private
+
+    # Checks if a given path corresponds to the base (English) strings file, en.lproj/Localizable.strings.
+    #
+    # @return [Boolean] true if path is the base strings file
+    def base_strings_file?(path:)
+      base_strings_path_components = Pathname.new(BASE_STRINGS_FILE).each_filename.to_a
+      path_components = Pathname.new(path).each_filename.to_a
+
+      base_strings_path_components == path_components.last(base_strings_path_components.length)
+    end
```

Comment:
I think in the future this method could be generalized to some `def matches_relative_path?` or something, in the collection of utility method that you're working on, so that every time we need a path comparison where you used to rely on `end_with?` in any of your rules you'd now be able to use that more resilient helper instead.

That could be for a future PR though.